ci-operator/step-registry/openshift/oc/certrotation/openshift-oc-certrotation-commands.sh

```diff
@@ -1,10 +1,56 @@
 #!/bin/bash
 set -xeuo pipefail
+
+start_date=$(date +"%Y-%m-%dT%H:%M:%S%:z")
+
+# ensure we're stable to start
 oc adm wait-for-stable-cluster --minimum-stable-period=5s
+
 # Let's start with the MCO cert rotation
 oc adm ocp-certificates regenerate-machine-config-server-serving-cert
-# A few preparatory rotations
+
+# A few preparatory rotations, these give us 30 days to complete the rotation after generating new roots of trust
 oc adm ocp-certificates regenerate-leaf -n openshift-config-managed secrets kube-controller-manager-client-cert-key kube-scheduler-client-cert-key
 oc adm ocp-certificates regenerate-leaf -n openshift-kube-apiserver-operator secrets node-system-admin-client
 oc adm ocp-certificates regenerate-leaf -n openshift-kube-apiserver secrets check-endpoints-client-cert-key control-plane-node-admin-client-cert-key  external-loadbalancer-serving-certkey internal-loadbalancer-serving-certkey kubelet-client localhost-recovery-serving-certkey localhost-serving-cert-certkey service-network-serving-certkey
 oc adm wait-for-stable-cluster
+
+# generate new roots of trust
+oc adm ocp-certificates regenerate-top-level -n openshift-kube-apiserver-operator secrets kube-apiserver-to-kubelet-signer kube-control-plane-signer  loadbalancer-serving-signer  localhost-serving-signer service-network-serving-signer
+oc -n openshift-kube-controller-manager-operator delete secrets/next-service-account-private-key
+oc -n openshift-kube-apiserver-operator delete secrets/next-bound-service-account-signing-key
+oc adm wait-for-stable-cluster
+
+# skip the AWS STS bound SA changes because we don't have the aws CLI or know where this info is stored.
+
+# generate new client certs for kcm and ks
+oc adm ocp-certificates regenerate-leaf -n openshift-config-managed secrets kube-controller-manager-client-cert-key kube-scheduler-client-cert-key
+
+# distribute trust across all known clients
+# update our local CA bundle so that when new serving certs are used for kube-apiserver we will trust them
+oc config refresh-ca-bundle
+# produce a new kubelet bootstrap kubeconfig (used to create the first CSR and establishes ca bundle)
+oc config new-kubelet-bootstrap-kubeconfig > bootstrap.kubeconfig
```

Comment:
it's this write here that is failing due to a lack of permissions .

Reply:
Should work to use `/tmp`